webui/react/src/e2e/tests/experimentList.spec.ts

```diff
@@ -311,4 +315,38 @@ test.describe('Experiment List', () => {
     await row.experimentActionDropdown.pause.pwLocator.click();
     await expect.soft((await row.getCellByColumnName('State')).pwLocator).toHaveText('paused');
   });
+
+  test('Datagrid Bulk Action', async () => {
+    // should probably go last/before move
+    await test.step('Kill', async () => {
+      type RowType = typeof projectDetailsPage.f_experimentList.dataGrid.rows;
+      const rows = [0, 1].map((idx) => {
+        return projectDetailsPage.f_experimentList.dataGrid.getRowByIndex(idx);
+      });
+      const expectStateForRow = async (state: string, row: RowType) => {
+        const stateColumn = await row.getCellByColumnName('State');
+        await expect(stateColumn.pwLocator).toHaveText(state);
+      };
+      await rows.reduce(async (memo, row) => {
+        await memo;
+        await expectStateForRow('paused', row);
+        await row.clickColumn('Select');
+      }, Promise.resolve());
+
+      await projectDetailsPage.f_experimentList.tableActionBar.actions.kill.select();
+
+      // TODO: modal component model assumes buttons are attached to form
```

Comment:
I'm glad you included the to do. would you be able to fix the model? I can pair if you want

Reply:
Having thought about this, my fix would be to remove it. I don't think the modal component object class is a useful abstraction. Page objects encapsulate and centralize functionality and locators, and objects in general create contracts that functions can use (i.e. any function that takes a `Modal` can call `Modal#open`).

If we take a step back and think about modals in general, what modal functionality could we encapsulate in a class? Currently we have the ability to open or close a modal. As seen in `BaseOverlay`, we can attempt to centralize this functionality by providing either a locator or a function to open the modal, but again:

* no code expects a `Modal`, so nothing outside of the test suites calls `Modal#open`
* we're either passing a selector in `clickThisComponentToOpen` or a function in `openMethod`, so we're not centralizing functionality or locators

You could argue that the overlay and modal components integrate visibility checks before further interactions but [playwright already checks for visibility and interactability when using a locator](https://playwright.dev/docs/actionability), so those checks seem to be redundant in this case. As such, I don't think we gain anything when writing `projectDetailsPage.f_experimentList.killModal.modalFooter.submit()` versus `projectDetailsPage.f_experimentList.killModal.confirm()`. In fact, i think we _lose_ something because the semantic actions of the modal are flattened -- i'm not submitting a form, i'm confirming a prompt.